azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/method_call.py

```diff
@@ -0,0 +1,39 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a class representing method calls that are received and responded to.
+"""
+
+
+class MethodCall(object):
+    """Represents a call to a device or module method.
+
+    :ivar str name: The name of the method being called.
+    :ivar payload: The payload being sent with the call.
+    :ivar int response_timeout: The time in seconds before a response to the call is invalid.
+    """
+
+    def __init__(self, name, payload, response_timeout):
+        """Initializer for a MethodCall.
+
+        :param str name: The name of the method being called.
+        :param payload: The payload being sent with the call.
+        :param int response_timeout: The time in seconds before a response to the call is invalid.
+        """
+        self._name = name
+        self._payload = payload
+        self._response_timeout = response_timeout
```

Comment:
>self._response_timeout = response_timeout [](start = 8, length = 41)

the device does not see this (only the service that has a response timeout and a connect timeout). There is however a request_id that is used to match request and response.

Reply:
Yeah, request ID is missing, it was a (rather egregious) oversight on my part.

---
In reply to: [266186098](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266186098) [](ancestors = 266186098)

Reply:
still need to remove the response_timeout property. not closed.

---
In reply to: [266595247](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266595247) [](ancestors = 266595247,266186098)